src/libs/ajax/analysis-providers/AnalysisProvider.ts

```diff
@@ -0,0 +1,62 @@
+import { DataClientDependencies, dataClientDependencies } from 'src/dependencies/data-client-resolver'
+import { Composable } from 'src/dependencies/dependency-core'
+import { GoogleWorkspaceInfo, isGoogleWorkspaceInfo, WorkspaceInfo } from 'src/libs/workspace-utils'
+import { AnalysisFile, getExtension, stripExtension } from 'src/pages/workspaces/workspace/analysis/file-utils'
+import { ToolLabel } from 'src/pages/workspaces/workspace/analysis/tool-utils'
+
+
+export type AnalysisProviderDeps = {
+  dataClients: Pick<DataClientDependencies, 'AzureStorage' | 'GoogleStorage'>
+}
+
+export interface AnalysisProviderContract {
+  listAnalyses: (workspaceInfo: WorkspaceInfo, signal?: AbortSignal) => Promise<AnalysisFile[]>
+  copyAnalysis: (
+      sourceWorkspace: WorkspaceInfo,
+      printName: string,
+      toolLabel: ToolLabel,
+      targetWorkspace: WorkspaceInfo,
+      newName: string,
+      signal?: AbortSignal
+  ) => Promise<void>
+}
+
+export type ComposableAnalysisProvider = Composable<AnalysisProviderDeps, AnalysisProviderContract>
+
+export const AnalysisProvider: ComposableAnalysisProvider = {
```

Comment:
this provider is implemented in a pending PR for IA team, shown here to have an more robust example of the proposed UI stack.